### PR TITLE
Login UI improv liu wu

### PR DESCRIPTION
**Pull request by WeMWish**

### CODE REVIEW

**Comment by Dsuzuki126** [comment]
A few things:
1) I recommend you guys write jest tests as manually testing is difficult and I dont think you wrote any (but if you did please show me where)
2) the password reset doesnt have any effect unless I copy and paste the url for the login page into another tab. Refreshing the page and clearing the cache doesn't work. 
3) when I click the "confirm" button there is still a prompt that says incorrect password 
4) the indexDB information doesnt reflect the change in real time hence having to go to another tab
5) small thing, in LoginJS.js on line 98 there is a 'g' missing in the console.log that may cause issues

**Comment by Dsuzuki126** [comment]
Another thing I discovered is that if I copy paste the url to another tab then go to a previous tab that I reset the password with and hard refresh the page then I can use the new PIN but the indexDB information is still wrong on the previous tab

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
A few things:
Thanks for removing the unnecessary Login alert
- I recommend you guys write jest tests as manually testing is difficult and I dont think you wrote any (but if you did please show me where); i think if you ran through the functionality you can find different errors that come up
- The password reset doesnt have any effect unless I copy and paste the url for the login page into another tab. Refreshing the page and clearing the cache doesn't work.
- when I click the "confirm" button there is still a prompt that says incorrect password
- the indexDB information doesnt reflect the change in real time hence having to go to another tab
- small thing, in LoginJS.js on line 98 there is a 'g' missing in the console.log that may cause issues
- the way you have implemented the password reset also makes it possible for a user to reset their password PIN to an empty string
- Another thing I discovered is that if I copy paste the url to another tab then go to a previous tab that I reset the password with and hard refresh the page then I can use the new PIN but the indexDB information is still wrong on the previous tab

**Comment by WeMWish** [comment]
- reset button is now fixed
- the incorrect password message is gone now when clicking confirm button
- line 98, console.log fixed
- the check password functionality is handled by other developers, and since the original code does include this functionality, we decided to leave it
- indexedDB should change dynamically now
- I added a jest test, but it fail to select the reset-password button on the login page. Since learning the jest test takes longer time than I expected and the buttons' functionality is pretty intuitive, to avoid affecting the next iteration, I would leave the test for now.

**Comment by Dsuzuki126** [comment]
Thanks for making the changes. 
I apologize as I think some of the indexDB not updating realtime may have been on my side.
There is one thing I am still worried about is that you can still click reset PIN and confirm with no PIN inputted so they could be entering the site with a PIN of empty string so maybe if you could check to make sure that the PIN isn't empty

**Comment by WeMWish** [comment]
the check for empty password functionality is updated. Thanks for the review!

**Comment by Dsuzuki126** [comment]
Noice I think its good now. Thanks guys

**Review by Dsuzuki126 — APPROVED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "cse110-sp21-group12",
```
Comment (k9kathyli) [inline package-lock.json:8]: what are the changes in is this file for?

File: source/Login/LoginJS.js [hunk 2]
```diff
@@ -84,11 +92,49 @@ function handleSignup(newUsername, newPassword) {
...
     sessionStorage.setItem('loggedIn', 'true');
     goHome();
 }
 
+/**
+ * handle reset password functionaliy of the associated
+ */
+/**
+ * handle the login button functionalities
+ */
```
Comment (k9kathyli) [inline source/Login/LoginJS.js:105]: weird commenting here

File: source/Login/LoginJS.js [hunk 3]
```diff
@@ -84,11 +92,49 @@ function handleSignup(newUsername, newPassword) {
...
+                // eslint-disable-next-line no-undef
+                updateSettings(userObject);
+                settingObj.password = passwordField.value;
+
+                // log the user in
+                sessionStorage.setItem('loggedIn', 'true');
+                goHome();
+                // // reset the button after clicked and update the settings
+                // loginButton.innerHTML = 'Sign-In';
+                // loginButton.addEventListener('click', handleLoginButton);
```
Comment (k9kathyli) [inline source/Login/LoginJS.js:128]: Commented code needed? If not, remove

File: source/tests/PuppTests.test.js [hunk 4]
```diff
@@ -874,4 +874,30 @@ describe('basic navigation for BJ', () => {
...
+        });
+        await page.waitForTimeout(300);
+
+        await page.$eval('#login-button', (button) => {
+            button.click();
+        });
+
+        const url2 = await page.evaluate(() => location.href);
+        expect(url2).toMatch(
+            'https://cse110-sp21-group12.github.io/cse110-sp21-group12/source/Index/Index.html'
```
Comment (k9kathyli) [inline source/tests/PuppTests.test.js:900]: wrong url

File: source/tests/PuppTests.test.js [hunk 5]
```diff
@@ -874,4 +874,30 @@ describe('basic navigation for BJ', () => {
         const currentTheme = await page.select('#themes', '#ECC7C7');
         expect(currentTheme.toString()).toMatch('#ECC7C7');
     });
+    it('Test51: Reset password, should be loggedIn state after resetting password', async () => {
+        jest.setTimeout(30000);
+
+        await page.goto(
+            'https://cse110-sp21-group12.github.io/cse110-sp21-group12/source/Login/Login.html'
```
Comment (k9kathyli) [inline source/tests/PuppTests.test.js:881]: wrong url

**Review by k9kathyli — CHANGES_REQUESTED** [review]
- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [x] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions 
- [ ] Code is commented/documented well
Please test locally and upload proof as discussed in Slack.

**Comment by WeMWish** [comment]
fix puppet test, style issue fixed

**Comment by PatriciaWuu** [comment]
I also removed all commented code as resolve #15 in this branch to wait for PR

**Comment by PatriciaWuu** [comment]
Screen Shot 2022-05-15 at 20 10 10

**Comment by PatriciaWuu** [comment]
Screen Shot 2022-05-16 at 15 48 47

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
make sure you pull all the changes from the main branch;

**Comment by Dsuzuki126** [comment]
just a reminder that after pulling from main, you may have conflicts and need to choose which code to use.  Make sure to resolve all change conflicts. It seems like the pupptests.test.js file some of your code mixed with the main branch code

File: source/DailyOverview/DailyOverview.html [hunk 6]
```diff
@@ -10,8 +10,13 @@
     <body>
         <div id="header">
             <h1 id="monthView">
-                <a href="../MonthlyOverview/MonthlyOverview.html" id="words">
-                </a>
```
Comment (Dsuzuki126) [inline source/DailyOverview/DailyOverview.html:14]: You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

File: source/MonthlyOverview/MonthlyOverview.html [hunk 7]
```diff
@@ -11,7 +11,12 @@
         <div id="header">
             <h1 id="year">
                 <h1 id="index">
-                    <a href="../YearlyOverview/YearlyOverview.html" id="back">
+                    <a
+                        href="../YearlyOverview/YearlyOverview.html"
+                        style="text-decoration: none"
+                        id="back"
```
Comment (Dsuzuki126) [inline source/MonthlyOverview/MonthlyOverview.html:17]: You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

File: source/YearlyOverview/YearlyOverview.html [hunk 8]
```diff
@@ -10,7 +10,13 @@
     <body>
         <div id="header">
             <h1 id="index">
-                <a href="../Index/Index.html" id="back"> Index</a>
+                <a
+                    href="../Index/Index.html"
+                    style="text-decoration: none"
+                    id="back"
```
Comment (Dsuzuki126) [inline source/YearlyOverview/YearlyOverview.html:16]: You may need to pull from main as I think someone maybe Hunter, changed the code so there isn't CSS with the HTML which you branch still has

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
just a reminder that after pulling from main, you may have conflicts and need to choose which code to use. Make sure to resolve all change conflicts. It seems like the pupptests.test.js file some of your code mixed with the main branch code

- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [x] Adequate test coverage
- [x]  Follows coding conventions
- [x] Code is commented/documented well

**Review by PatriciaWuu — APPROVED** [review]
resolved

File: source/tests/PuppTests.test.js [hunk 9]
```diff
@@ -54,7 +54,7 @@ describe('basic navigation for BJ', () => {
         await page.waitForTimeout(1000);
     });
 
-    it('Test1: Initial Home Page - Shows create your login ', async () => {
```
Comment (Dsuzuki126) [inline source/tests/PuppTests.test.js:57]: I tried running the Puppeteer tests with your code and it failed a lot so maybe you can take time to figure out what is going on or just send a screenshot that it works on your machine
image

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
- [ ] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions
- [ ] Code is commented/documented well

**Review by Dsuzuki126 — APPROVED** [review]

**Review by Dsuzuki126 — APPROVED** [review]
I ran the tests on my computer and they all passed. Just make sure they pass for you guys.
- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well

**Review by k9kathyli — APPROVED** [review]

**Review by Dsuzuki126 — APPROVED** [review]
Doesn't look like any of the files changed for me. 
All the tests passed as well.

- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well

**Review by ManitKSoni — APPROVED** [review]
- [x] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [x] Follows coding conventions
- [x] Code is commented/documented well

**Review by Dsuzuki126 — APPROVED** [review]
Nice job fixing the double error thing
- [ ] Straightforward implementation
- [ ] Good design
- [ ] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [ ] Adequate test coverage
- [ ] Follows coding conventions
- [ ] Code is commented/documented well

**Pull request merged**